fastplotlib/layouts/_gridplot.py

```diff
@@ -110,127 +117,177 @@ def __init__(
             raise ValueError("Number of cameras does not match the number of subplots")
 
         # create the cameras
-        self._cameras = np.empty(self.shape, dtype=object)
+        subplot_cameras = np.empty(self.shape, dtype=object)
         for i, j in product(range(self.shape[0]), range(self.shape[1])):
-            self._cameras[i, j] = create_camera(camera_type=cameras[i, j])
+            subplot_cameras[i, j] = create_camera(camera_type=cameras[i, j])
 
-        if controller_ids is None:
-            # individual controller for each subplot
-            controller_ids = np.arange(self.shape[0] * self.shape[1]).reshape(
-                self.shape
-            )
+        # if controller instances have been specified for each subplot
+        if controllers is not None:
+
+            # one controller for all subplots
+            if isinstance(controllers, pygfx.Controller):
+                controllers = [controllers] * len(self)
+                # subplot_controllers[:] = controllers
+                # # subplot_controllers = np.asarray([controllers] * len(self), dtype=object)
 
-        elif isinstance(controller_ids, str):
-            if controller_ids == "sync":
-                controller_ids = np.zeros(self.shape, dtype=int)
+            # individual controller instance specified for each subplot
             else:
-                raise ValueError(
-                    f"`controller_ids` must be one of 'sync', an array/list of subplot names, or an array/list of "
-                    f"integer ids. See the docstring for more details."
-                )
+                # I found that this is better than list(*chain(<list/array>)) because chain doesn't give the right
+                # result we want for arrays
+                for item in controllers:
+                    if isinstance(item, pygfx.Controller):
+                        pass
+                    elif all(isinstance(c, pygfx.Controller) for c in item):
+                        pass
+                    else:
+                        raise TypeError(
+                            "controllers argument must be a single pygfx.Controller instance of a Iterable of "
+                            "pygfx.Controller instances"
```

Comment:
"controllers argument must be a single pygfx.Controller instance or an Iterable of
 pygfx.Controller instances"